sdk/inc/azure/core/az_mqtt5_rpc_client_codec.h

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Definition of #az_mqtt5_rpc_client_codec.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_MQTT5_RPC_CLIENT_CODEC_H
+#define _az_MQTT5_RPC_CLIENT_CODEC_H
+
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief MQTT5 RPC Client Codec options.
+ *
+ */
+typedef struct
+{
+  /**
+   * @brief Timeout in seconds for subscribing (must be > 0).
+   */
+  int32_t subscribe_timeout_in_seconds;
+
+  /**
+   * @brief Timeout in seconds for publishing (must be > 0).
+   */
+  int32_t publish_timeout_in_seconds;
+
+  /**
+   * @brief The topic format to use for the subscription topic.
+   *
+   *
+   * @note Can include {name} for command name, {serviceId} for model id, {executorId} for the
+   * server's client_id, and/or {invokerClientId} for the client's client_id.
+   */
+  az_span subscription_topic_format;
+
+  /**
+   * @brief The topic format to use for the request topic.
+   *
+   * @note Can include {name} for command name, {serviceId} for model id, and/or {executorId} for
```

Comment:
If there are other requirements now with the slashes, can you include those here as well?